res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,741 @@
+        engine.scratchTick(deckNum, newVal);
+    } else { // fallback
+        engine.setValue(group, "jog", newVal * this.bendScale);
+    }
+};
+
+
+PioneerDDJ400.jogSearch = function(_channel, _control, value, _status, group) {
+    // "highspeed" (scaleup value) pitch bend
+    var newVal = (value - 64) * this.highspeedScale;
+    engine.setValue(group, "jog", newVal);
+};
+
+PioneerDDJ400.jogTouch = function(channel, _control, value) {
+    var deckNum = channel + 1;
+
+    // skip scratchmode if we adjust the loop points
+    if (PioneerDDJ400.loopAdjustIn[channel] || PioneerDDJ400.loopAdjustOut[channel]) {
+        return;
+    }
+
+    // on touch jog with vinylmode enabled -> enable scratchmode
+    if (value !== 0 && this.vinylMode) {
+        engine.scratchEnable(deckNum, 720, 33+1/3, this.alpha, this.beta);
+    } else {
+        // on release jog (value === 0) disable pitch bend mode or scratch mode
+        engine.scratchDisable(deckNum);
+    }
+};
+
+//
+// SHIFT
+//
+
+PioneerDDJ400.shiftPressed = function(channel, _control, value) {
+    this.shiftState[channel] = value;
+    this.performancePads.toggleShiftLights(channel, value);
+};
+
+//
+// Pad mode buttons
+//
+
+PioneerDDJ400.setPadmode = function(channel, control, value) {
+    if (value === 0x7F) {
+        PioneerDDJ400.performancePads.state[channel] = control;
+    }
+};
+
+//
+// Beat Jump mode
+//
+
+PioneerDDJ400.beatjumpPadPressed = function(_channel, control, value, _status, group) {
+    if (value === 0) {
+        return;
+    }
+    engine.setValue(group, "beatjump_size", Math.abs(PioneerDDJ400.beatjumpPad[control]));
+    engine.setValue(group, "beatjump", PioneerDDJ400.beatjumpPad[control]);
+};
+
+PioneerDDJ400.beatjumpShiftUp = function(_channel, control, value, _status, group) {
+    if (value === 0 || PioneerDDJ400.beatjumpPad[0x21] * 16 > 16) {
+        return;
+    }
+    Object.keys(PioneerDDJ400.beatjumpPad).forEach(function(pad) {
+        PioneerDDJ400.beatjumpPad[pad] = PioneerDDJ400.beatjumpPad[pad] * 16;
+    });
+    engine.setValue(group, "beatjump_size", PioneerDDJ400.beatjumpPad[0x21]);
+};
+
+PioneerDDJ400.beatjumpShiftDown = function(_channel, control, value, _status, group) {
+    if (value === 0 || PioneerDDJ400.beatjumpPad[0x21] / 16 < 1/16) {
+        return;
+    }
+    Object.keys(PioneerDDJ400.beatjumpPad).forEach(function(pad) {
+        PioneerDDJ400.beatjumpPad[pad] = PioneerDDJ400.beatjumpPad[pad] / 16;
+    });
+    engine.setValue(group, "beatjump_size", PioneerDDJ400.beatjumpPad[0x21]);
+};
+
+//
+// Sampler mode
+//
+
+PioneerDDJ400.samplerPlayOutputCallbackFunction = function(value, group, _control) {
+    if (value === 1) {
+        var curPad = group.match(/^\[Sampler(\d+)\]$/)[1];
+        PioneerDDJ400.startSamplerBlink((0x97 + (curPad > 8 ? 2 : 0)), (0x30 + ((curPad > 8 ? curPad-8 : curPad)-1)), group);
+    }
+};
+
+PioneerDDJ400.samplerModePadPressed = function(_channel, _control, value, _status, group) {
```

Comment:
```suggestion
PioneerDDJ400.samplerPadPressed = function(_channel, _control, value, _status, group) {
```
I got quite confused thinking this function was mapped to the button that switches to the sampler mode.